sdk/src/args.rs

```diff
@@ -1,98 +1,369 @@
     pub quic_receive_window: u64,
 
-    /// The optional response buffer size for QUIC.
-    #[arg(long, default_value = "1048576")]
+    /// The optional response buffer size for QUIC
     pub quic_response_buffer_size: u64,
 
-    /// The optional keep alive interval for QUIC.
-    #[arg(long, default_value = "5000")]
+    /// The optional keep alive interval for QUIC
     pub quic_keep_alive_interval: u64,
 
-    /// The optional maximum idle timeout for QUIC.
-    #[arg(long, default_value = "10000")]
+    /// The optional maximum idle timeout for QUIC
     pub quic_max_idle_timeout: u64,
 
-    /// Flag to enable certificate validation for QUIC.
-    #[arg(long, default_value = "false")]
+    /// Flag to enable certificate validation for QUIC
     pub quic_validate_certificate: bool,
 }
+
+impl Args {
+    pub fn get_server_address(&self) -> Option<String> {
+        match self.transport.as_str() {
+            QUIC_TRANSPORT => Some(
+                self.quic_server_address
+                    .split(':')
+                    .next()
+                    .unwrap()
+                    .into(),
+            ),
+            HTTP_TRANSPORT => Some(
+                self.http_api_url
+                    .clone()
+                    .replace("http://", "")
+                    .replace("localhost", "127.0.0.1")
+                    .split(':')
+                    .next()
+                    .unwrap()
+                    .into(),
+            ),
+            TCP_TRANSPORT => Some(
+                self.tcp_server_address
+                    .split(':')
+                    .next()
+                    .unwrap()
+                    .into(),
+            ),
+            _ => None,
+        }
+    }
+}
+
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            transport: "tcp".to_string(),
+            encryption_key: "".to_string(),
+            http_api_url: "http://localhost:3000".to_string(),
+            http_retries: 3,
+            tcp_server_address: "127.0.0.1:8090".to_string(),
+            tcp_reconnection_retries: 3,
+            tcp_reconnection_interval: 1000,
+            tcp_tls_enabled: false,
+            tcp_tls_domain: "localhost".to_string(),
+            quic_client_address: "127.0.0.1:0".to_string(),
+            quic_server_address: "127.0.0.1:8080".to_string(),
+            quic_server_name: "localhost".to_string(),
+            quic_reconnection_retries: 3,
+            quic_reconnection_interval: 1000,
+            quic_max_concurrent_bidi_streams: 10000,
+            quic_datagram_send_buffer_size: 100000,
+            quic_initial_mtu: 1200,
+            quic_send_window: 100000,
+            quic_receive_window: 100000,
+            quic_response_buffer_size: 1048576,
+            quic_keep_alive_interval: 5000,
+            quic_max_idle_timeout: 10000,
+            quic_validate_certificate: false,
+        }
+    }
+}
+
+impl From<Vec<ArgsOptional>> for Args {
```

Comment:
Why here's Vec<ArgsOptional> ?

Reply:
I have not implemented it yet but this will be used for merging args provided via the cli with "args" from the config file